src/bm_sim/actions.cpp

```diff
@@ -283,6 +283,9 @@ void
 ActionPrimitiveCall::execute(ActionEngineState *state,
                              const ActionParam *args) const {
   // TODO(unknown): log source info?
+  if (source_info != nullptr) {
```

Comment:
is there really a need for a conditional here? it seems to be you can just call `set_source_info` without checking for null.
it's even worse than that actually. If it's null and you don't set it, isn't there a chance of a "stale" / incorrect source info (from a different primitive call) being displayed?

Reply:
You are right, it seems that can just call set_source_info. I have made a change according to your comment.